src/mapping/query/function/sha2.rs

```diff
@@ -0,0 +1,156 @@
+use super::prelude::*;
+use digest::DynDigest;
+use sha2::{Digest, Sha224, Sha256, Sha384, Sha512, Sha512Trunc224, Sha512Trunc256};
+
+#[derive(Debug)]
+pub(in crate::mapping) struct Sha2Fn {
+    query: Box<dyn Function>,
+    variant: Option<Box<dyn Function>>,
+}
+
+impl Sha2Fn {
+    #[cfg(test)]
+    pub(in crate::mapping) fn new(query: Box<dyn Function>, variant: Option<&str>) -> Self {
+        let variant = variant.map(|v| Box::new(Literal::from(Value::from(v))) as _);
+
+        Self { query, variant }
+    }
+}
+
+impl Function for Sha2Fn {
+    fn execute(&self, ctx: &Event) -> Result<Value> {
+        let bytes = required!(ctx, self.query, Value::Bytes(v) => v);
+
+        let variant = optional!(ctx, self.variant, Value::Bytes(v) => v)
+            .map(|v| String::from_utf8_lossy(&v).into_owned());
```

Comment:
Can the `.into_owned` be dropped to avoid the effective clone here?

Reply:
Not as-is (since the owned `v` is dropped at the end), but I made some changes to remove the need for comparing strings, comparing bytes instead.